lib/cloud/clients.go

```diff
@@ -630,6 +645,21 @@ func (c *cloudClients) initGCPGKEClient(ctx context.Context) (gcp.GKEClient, err
 	return gcpGKE, nil
 }
 
+func (c *cloudClients) initGCPInstancesClient(ctx context.Context) (gcp.InstancesClient, error) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	if c.gcpInstances != nil { // If some other thread already got here first.
```

Comment:
I don't get it. What do you mean by another thread? You acquire mutex first. If you meant "return if has been already initialized", then you can use `sync.Once`